### ISSUE TITLE
Save entry -> (No changes need to be saved)

### ISSUE THREAD

**Issue by entropiece** [punchagan/org2blog#179]
When trying to publish a test post, after invoking `org2blog/wp-post-buffer-and-publish` I get 

`Save entry? (y or n)`.  Pressing `y` or 'n' does not do anyhting useful and the ony way to go back is with `C-g`. In `*Messages*` I can see `(No changes need to be saved)`. 

Does anybody have an idea of what's going on?

**Comment by DiogoFerrari** [comment]
Try to save the file in your drive first, and them post it. See if that works.

**Comment by punchagan** [comment]
What does `(buffer-modified-p)` return when evaluated in such a buffer? You can evaluate it by `M-S-: (buffer-modified-p) RET`.

**Comment by entropiece** [comment]
@punchagan It returns `t`. So I tried the suggestion from @DiogoFerrari and this is in fact the problem. It seems you cant publish a post from a buffer, it has to be a saved file instead. 
Thanks for the help

**Comment by punchagan** [comment]
If it return `t`, you should actually be prompted for a location to save the buffer in.  Just like you are prompted, when you hit `C-x C-s` in a new buffer.

**Comment by treese** [comment]
I ran into the same problem. This seems to be a bad interaction between `org2blog/wp-kill-buffer-hook`, which is what prompts to save an entry, and the changes in org-mode added in commit 0e01e3eb13ab4045be51395f98c2d20d5dbaee1d on July 25. The code in `org-element-parse-secondary-string` clones all local variables from the buffer to the temporary buffer that it uses, which includes `kill-buffer-hook`, which has been set as a buffer local variable at that point. So the message is coming when the tempoary buffer is killed.

A stack trace shows that this happens during the call to `org-export-get-environment` in `org2blog/wp--parse-buffer-entry`. This problem doesn't seem to occur with the standard org-mode exporters. In the current version of `ox-publish.el`, a similar call to `org-export-get-environment` is wrapped with a call to `org-export-with-buffer-copy`.

I modified `org2blog/wp--parse-buffer-entry` to make the copy (and just do it once, since it was getting the environment twice), and it works much better now. Submitted a pull request with the changes.
